gateway/gateway.go

```diff
@@ -1368,7 +1368,7 @@ func (gateway *HandleT) StartWebHandler(ctx context.Context) error {
 
 	srvMux := mux.NewRouter()
 	srvMux.Use(
-		middleware.StatMiddleware(ctx),
+		middleware.StatMiddleware(ctx, srvMux),
```

Comment:
the idea of passing a `router` to it's own middleware func as an argument looks weird to me. But, I am unable to think of an alternative to `router.Match`.